src/hotspot/share/code/nmethod.cpp

```diff
@@ -1236,35 +1227,51 @@ nmethod::nmethod(
     _orig_pc_offset          = 0;
     _deopt_handler_offset    = 0;
     _deopt_mh_handler_offset = 0;
+    _unwind_handler_offset   = 0;
     _gc_epoch                = CodeCache::gc_epoch();
 
-    _consts_offset           = content_offset()      + code_buffer->total_offset_of(code_buffer->consts());
-    _stub_offset             = content_offset()      + code_buffer->total_offset_of(code_buffer->stubs());
-    _oops_offset             = data_offset();
-    _metadata_offset         = _oops_offset          + align_up(code_buffer->total_oop_size(), oopSize);
-    _scopes_data_offset      = _metadata_offset      + align_up(code_buffer->total_metadata_size(), wordSize);
-    _scopes_pcs_offset       = _scopes_data_offset;
-    _dependencies_offset     = _scopes_pcs_offset;
-    _handler_table_offset    = _dependencies_offset;
+    int skipped_insts_size   = code_buffer->total_skipped_instructions_size();
+#ifdef ASSERT
+    assert(((skipped_insts_size >> 16) == 0), "size is bigger than 64Kb: %d", skipped_insts_size);
+
+    // SECT_CONSTS is first in code buffer so the offset should be 0.
+    int consts_offset = code_buffer->total_offset_of(code_buffer->consts());
+    assert(consts_offset == 0, "const_offset: %d", consts_offset);
+#endif
```

Comment:
```suggestion
```

Reply:
I assume you are suggesting to remove `#ifdef ASSERT`.
Done.